### ISSUE TITLE
The Forms folder structure is not sanitized which can lead to synchronization issues when uploading files

### ISSUE THREAD

**Issue by stefan0xC** [nextcloud/forms#2409]
**Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!**

**Describe the bug**
Uploaded files are seemingly saved in a folder structure `Forms/{id of form} - {title of form}/{id of submission}/{id of question} - {title of question}` but the titles do not seem to get trimmed in regards to trailing white spaces, which results in syncing problems (at least on Windows 11 with NTFS).

**To Reproduce**

1. Go to Forms and create a new form
2. Call it `test ` with a trailing whitespace
3. Add a file upload option and call it `test ` as well
4. Upload a file

**Expected behavior**
I expect the fields that are used for filenames to be trimmed in regard to whitespace and maybe also sanitized regarding other invalid characters like `:`

**Screenshots**
![Image](https://github.com/user-attachments/assets/8988ea39-9085-4a40-bcc1-9d28364b628b)

**Nextcloud (please complete the following information):**

-   Nextcloud-Version: 30.0.2
-   Forms-Version: 4.3.3

**Desktop (please complete the following information):**

-   OS: Windows 11 with NTFS file system
- Nexcloud Desktop v3.14.3

**Comment by Koc** [comment]
I've tested it few scenarios:
1. create form with trailing `:`
2. create file question with trailing `:`
3. upload file on Linux (devcontainer)

everything works fine
![Image](https://github.com/user-attachments/assets/7d729977-b11c-49a9-bd96-7f144b758881)

1. create form with trailing `: `
2. create file question with trailing `: `
3. upload file on Linux (devcontainer)

the same
![Image](https://github.com/user-attachments/assets/0f442a1d-fb04-4cd6-85b6-d8f713f26a53)

We're already using [`normalizeFileName`](https://github.com/nextcloud/forms/blob/main/lib/Service/FormsService.php#L778-L778) which uses `\OCP\Constants::FILENAME_INVALID_CHARS` under the hood. And as far I can see from the constant [description](FILENAME_INVALID_CHARS) - we aren't supporting Windows.

**Comment by Chartman123** [comment]
@Koc yes that's right, at least server side. But the official sync client sure does support Windows. Perhaps we can have a look how Server/Files solves that problem?

**Comment by Koc** [comment]
Finally I was able to reproduce this issue. [Fix](https://github.com/nextcloud/forms/pull/2513)
